include/mapped_soa.h

```diff
@@ -85,6 +85,10 @@ class MappedSoA {
     key_to_index_.erase(key);
   }
 
+  void copy(size_t indexFrom, size_t indexTo) {
```

Comment:
One other thought, should we have another function that copies by key? Should we rename this and swap to copy_by_index and swap_by_index and have copy and swap take key by default? (or copy_by_key).